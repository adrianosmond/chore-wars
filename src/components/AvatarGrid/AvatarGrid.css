.avatar-grid {
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: repeat(auto-fit, minmax(100px, 1fr));
  list-style-type: none;
  padding-left: 0;
}

.avatar-grid__option {
  position: relative;
}

@media screen and (min-width: 37.5em) {
  .avatar-grid {
    grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
  }
}