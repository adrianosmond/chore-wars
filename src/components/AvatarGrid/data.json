{
  "skinColor": [
    "Pale",
    "Light",
    "Tanned",
    "Yellow",
    "Brown",
    "DarkBrown",
    "Black"
  ],
  "topType": [
    "NoHair",
    "Eyepatch",
    "Hat",
    "Hijab",
    "Turban",
    "WinterHat1",
    "WinterHat2",
    "WinterHat3",
    "WinterHat4",
    "LongHairBigHair",
    "LongHairBob",
    "LongHairBun",
    "LongHairCurly",
    "LongHairCurvy",
    "LongHairDreads",
    "LongHairFrida",
    "LongHairFro",
    "LongHairFroBand",
    "LongHairNotTooLong",
    "LongHairShavedSides",
    "LongHairMiaWallace",
    "LongHairStraight",
    "LongHairStraight2",
    "LongHairStraightStrand",
    "ShortHairDreads01",
    "ShortHairDreads02",
    "ShortHairFrizzle",
    "ShortHairShaggyMullet",
    "ShortHairShortCurly",
    "ShortHairShortFlat",
    "ShortHairShortRound",
    "ShortHairShortWaved",
    "ShortHairSides",
    "ShortHairTheCaesar",
    "ShortHairTheCaesarSidePart"
  ],
  "hatColor": [
    "Black",
    "Blue01",
    "Blue02",
    "Blue03",
    "Gray01",
    "Gray02",
    "Heather",
    "PastelBlue",
    "PastelGreen",
    "PastelOrange",
    "PastelRed",
    "PastelYellow",
    "Pink",
    "Red",
    "White"
  ],
  "hairColor": [
    "Auburn",
    "Black",
    "Blonde",
    "BlondeGolden",
    "Brown",
    "BrownDark",
    "PastelPink",
    "Platinum",
    "Red",
    "SilverGray"
  ],
  "facialHairType": [
    "Blank",
    "BeardMedium",
    "BeardLight",
    "BeardMagestic",
    "MoustacheFancy",
    "MoustacheMagnum"
  ],
  "facialHairColor": [
    "Auburn",
    "Black",
    "Blonde",
    "BlondeGolden",
    "Brown",
    "BrownDark",
    "Platinum",
    "Red"
  ],
  "accessoriesType": [
    "Blank",
    "Kurt",
    "Prescription01",
    "Prescription02",
    "Round",
    "Sunglasses",
    "Wayfarers"
  ],
  "clotheType": [
    "BlazerShirt",
    "BlazerSweater",
    "CollarSweater",
    "GraphicShirt",
    "Hoodie",
    "Overall",
    "ShirtCrewNeck",
    "ShirtScoopNeck",
    "ShirtVNeck"
  ],
  "clotheColor": [
    "Black",
    "Blue01",
    "Blue02",
    "Blue03",
    "Gray01",
    "Gray02",
    "Heather",
    "PastelBlue",
    "PastelGreen",
    "PastelOrange",
    "PastelRed",
    "PastelYellow",
    "Pink",
    "Red",
    "White"
  ]
}