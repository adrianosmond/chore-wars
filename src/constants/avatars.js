export const avatarOptions = {
  skinColor: [
    'Pale',
    'Light',
    'Tanned',
    'Yellow',
    'Brown',
    'DarkBrown',
    'Black',
  ],
  topType: [
    'NoHair',
    'Eyepatch',
    'Hat',
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
    'LongHairBigHair',
    'LongHairBob',
    'LongHairBun',
    'LongHairCurly',
    'LongHairCurvy',
    'LongHairDreads',
    'LongHairFrida',
    'LongHairFro',
    'LongHairFroBand',
    'LongHairNotTooLong',
    'LongHairShavedSides',
    'LongHairMiaWallace',
    'LongHairStraight',
    'LongHairStraight2',
    'LongHairStraightStrand',
    'ShortHairDreads01',
    'ShortHairDreads02',
    'ShortHairFrizzle',
    'ShortHairShaggyMullet',
    'ShortHairShortCurly',
    'ShortHairShortFlat',
    'ShortHairShortRound',
    'ShortHairShortWaved',
    'ShortHairSides',
    'ShortHairTheCaesar',
    'ShortHairTheCaesarSidePart',
  ],
  hatColor: [
    'Black',
    'Blue01',
    'Blue02',
    'Blue03',
    'Gray01',
    'Gray02',
    'Heather',
    'PastelBlue',
    'PastelGreen',
    'PastelOrange',
    'PastelRed',
    'PastelYellow',
    'Pink',
    'Red',
    'White',
  ],
  hairColor: [
    'Auburn',
    'Black',
    'Blonde',
    'BlondeGolden',
    'Brown',
    'BrownDark',
    'PastelPink',
    'Platinum',
    'Red',
    'SilverGray',
  ],
  facialHairType: [
    'Blank',
    'BeardMedium',
    'BeardLight',
    'BeardMagestic',
    'MoustacheFancy',
    'MoustacheMagnum',
  ],
  facialHairColor: [
    'Auburn',
    'Black',
    'Blonde',
    'BlondeGolden',
    'Brown',
    'BrownDark',
    'Platinum',
    'Red',
  ],
  accessoriesType: [
    'Blank',
    'Kurt',
    'Prescription01',
    'Prescription02',
    'Round',
    'Sunglasses',
    'Wayfarers',
  ],
  clotheType: [
    'BlazerShirt',
    'BlazerSweater',
    'CollarSweater',
    'Hoodie',
    'Overall',
    'ShirtCrewNeck',
    'ShirtScoopNeck',
    'ShirtVNeck',
  ],
  clotheColor: [
    'Black',
    'Blue01',
    'Blue02',
    'Blue03',
    'Gray01',
    'Gray02',
    'Heather',
    'PastelBlue',
    'PastelGreen',
    'PastelOrange',
    'PastelRed',
    'PastelYellow',
    'Pink',
    'Red',
    'White',
  ],
};


export const editorOrder = [
  'skinColor',
  'topType',
  'hairColor',
  'hatColor',
  'facialHairType',
  'facialHairColor',
  'accessoriesType',
  'clotheType',
  'clotheColor',
];

export const labels = {
  skinColor: 'Skin Colour',
  topType: 'Hair or Headware',
  hairColor: 'Hair Colour',
  hatColor: 'Headware Colour',
  facialHairType: 'Facial Hair Type',
  facialHairColor: 'Facial Hair Colour',
  accessoriesType: 'Glasses',
  clotheType: 'Clothing Type',
  clotheColor: 'Clothing Colour',
};

export const isEditable = {
  skinColor: () => true,
  topType: () => true,
  hairColor: options => [
    'NoHair',
    'Eyepatch',
    'Hat',
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
    'LongHairFrida',
    'LongHairShavedSides',
  ].indexOf(options.topType) < 0,
  hatColor: options => [
    'Hijab',
    'Turban',
    'WinterHat1',
    'WinterHat2',
    'WinterHat3',
    'WinterHat4',
  ].indexOf(options.topType) >= 0,
  facialHairType: options => options.topType !== 'Hijab',
  facialHairColor: options => isEditable.facialHairType(options) && options.facialHairType !== 'Blank',
  accessoriesType: () => true,
  clotheType: () => true,
  clotheColor: options => !options.clotheType.startsWith('Blazer'),
};

export const DefaultAvatar = {
  accessoriesType: 'Blank',
  avatarStyle: 'Transparent',
  clotheColor: 'Blue03',
  clotheType: 'BlazerShirt',
  eyeType: 'Default',
  eyebrowType: 'Default',
  facialHairColor: 'BrownDark',
  facialHairType: 'Blank',
  hairColor: 'BrownDark',
  mouthType: 'Twinkle',
  skinColor: 'Light',
  topType: 'LongHairStraight',
};
