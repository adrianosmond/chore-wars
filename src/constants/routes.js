export const LOGIN = '/'
export const CHORES = '/chores'
export const NEW_CHORE = '/chores/new'
