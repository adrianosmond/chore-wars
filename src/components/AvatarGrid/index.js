import React from 'react';
import Avatar from 'avataaars';

import data from './data.json';

import './index.css';

const AvatarGrid = ({ editing, avatar, updateAvatar }) =>
  <ul className="avatar-grid">
    {data[editing].map((option, idx) => {
      const avatarProps = {
        ...avatar,
        [editing]: option,
      };
      return (
        <li className="avatar-grid__option" key={idx} onClick={() => {
          updateAvatar(editing, option);
        }}>
          <Avatar { ...avatarProps} style={{ width: '100%', height: '100%' }} />
        </li>
      );
    })}
  </ul>;

export default AvatarGrid;
