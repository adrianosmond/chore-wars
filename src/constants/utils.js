import { TIME_UNIT } from './constants'
const sortByCurrentPoints = (a, b) => b.currentPoints - a.currentPoints

const processChore = (chore) => {
  const timeSinceChore = (new Date().getTime() - chore.lastDone) / TIME_UNIT
  const timeRemaining = timeSinceChore - chore.frequency
  const percentage = Math.min(100 * timeSinceChore / chore.frequency, 100)
  const multiplier = 1 + (1 / chore.frequency)
  const due = chore.lastDone + (chore.frequency * TIME_UNIT)
  return {
    currentPoints: Math.round(percentage / 100 * chore.pointsPerTime * Math.pow(multiplier, timeRemaining)),
    percentage,
    due
  }
}

export { sortByCurrentPoints, processChore }
