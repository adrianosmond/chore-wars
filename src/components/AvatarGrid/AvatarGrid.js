import React from 'react';
import { Link } from 'react-router-dom';
import Avatar from 'avataaars';

import { avatarOptions } from '../../constants/avatars';
import * as routes from '../../constants/routes';

import './AvatarGrid.css';

const AvatarGrid = ({ editing, avatar, onSelect }) =>
  <ul className="avatar-grid">
    {avatarOptions[editing].map((option, idx) => {
      const avatarProps = {
        ...avatar,
        [editing]: option,
      };
      return (
        <li className="avatar-grid__option" key={idx}>
          <Link to={routes.EDIT_PROFILE} onClick={() => { onSelect(option); }}>
            <Avatar { ...avatarProps} style={{ width: '100%', height: '100%' }} />
          </Link>
        </li>
      );
    })}
  </ul>;

export default AvatarGrid;
