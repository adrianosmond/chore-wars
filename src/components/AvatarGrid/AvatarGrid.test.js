import React from 'react';
import { shallow } from 'enzyme';
import { DefaultAvatar } from 'constants/avatars';

import AvatarGrid from './AvatarGrid';

describe('AvatarGrid', () => {
  it('Renders', () => {
    const component = shallow(<AvatarGrid editing="topType" avatar={DefaultAvatar} onSelect={jest.fn} />);
    expect(component).toMatchSnapshot();
  });
});
