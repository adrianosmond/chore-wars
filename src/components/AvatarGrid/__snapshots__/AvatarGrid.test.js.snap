// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGrid Renders 1`] = `
<ul
  className="avatar-grid"
>
  <li
    className="avatar-grid__option"
    key="NoHair"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="NoHair"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="Eyepatch"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="Eyepatch"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="Hat"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="Hat"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="Hijab"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="Hijab"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="Turban"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="Turban"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="WinterHat1"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="WinterHat1"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="WinterHat2"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="WinterHat2"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="WinterHat3"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="WinterHat3"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="WinterHat4"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="WinterHat4"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairBigHair"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairBigHair"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairBob"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairBob"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairBun"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairBun"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairCurly"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairCurly"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairCurvy"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairCurvy"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairDreads"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairDreads"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairFrida"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairFrida"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairFro"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairFro"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairFroBand"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairFroBand"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairNotTooLong"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairNotTooLong"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairShavedSides"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairShavedSides"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairMiaWallace"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairMiaWallace"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairStraight"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairStraight2"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraight2"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="LongHairStraightStrand"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="LongHairStraightStrand"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="ShortHairDreads01"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="ShortHairDreads01"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="ShortHairDreads02"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="ShortHairDreads02"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="ShortHairFrizzle"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="ShortHairFrizzle"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="ShortHairShaggyMullet"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="ShortHairShaggyMullet"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="ShortHairShortCurly"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="ShortHairShortCurly"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="ShortHairShortFlat"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="ShortHairShortFlat"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="ShortHairShortRound"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="ShortHairShortRound"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="ShortHairShortWaved"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="ShortHairShortWaved"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="ShortHairSides"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="ShortHairSides"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="ShortHairTheCaesar"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="ShortHairTheCaesar"
      />
    </Link>
  </li>
  <li
    className="avatar-grid__option"
    key="ShortHairTheCaesarSidePart"
  >
    <Link
      onClick={[Function]}
      to="/edit-profile"
    >
      <AvatarComponent
        accessoriesType="Blank"
        avatarStyle="Transparent"
        clotheColor="Blue03"
        clotheType="BlazerShirt"
        eyeType="Default"
        eyebrowType="Default"
        facialHairColor="BrownDark"
        facialHairType="Blank"
        hairColor="BrownDark"
        mouthType="Twinkle"
        skinColor="Light"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
        topType="ShortHairTheCaesarSidePart"
      />
    </Link>
  </li>
</ul>
`;
